mmdet3d/models/data_preprocessors/data_preprocessor.py

```diff
@@ -378,10 +381,50 @@ def voxelize(self, points: List[torch.Tensor]) -> Dict[str, torch.Tensor]:
                 coors.append(res_coors)
             voxels = torch.cat(points, dim=0)
             coors = torch.cat(coors, dim=0)
+        elif self.voxel_type == 'cylindrical':
+            voxels, coors = [], []
+            for i, (res, data_sample) in enumerate(zip(points, data_samples)):
+                rho = torch.sqrt(res[:, 0]**2 + res[:, 1]**2)
+                phi = torch.atan2(res[:, 1], res[:, 0]) / torch.pi * 180
+                polar_res = torch.stack((rho, phi, res[:, 2]), dim=-1)
+                # TODO: implement cylindrical voxelization in voxel_layer
+                min_bound = polar_res.new_tensor(
+                    self.voxel_layer.point_cloud_range[:3])
+                max_bound = polar_res.new_tensor(
+                    self.voxel_layer.point_cloud_range[3:])
+                polar_res = torch.clamp(polar_res, min_bound, max_bound)
```

Comment:
Both `torch.clamp` and `torch.clip` do not support Tenor input for `min` and `max` value when PyTorch < 1.9.0.
See the [api](https://pytorch.org/docs/1.8.0/generated/torch.clamp.html#torch.clamp) docs.

Reply:
Keep the current changes and optimize the voxelization op later.

Reply:
We should add comment above to remind users that currently we only support PyTorch version >= 1.9.0, and solve it in the future.

Reply:
agree